### PR TITLE
sandboxfs: init at 0.2.0

### PR DESCRIPTION
**Pull request by jeremyschlatter**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] ~macOS~ doesn't build on macOS b/c of fuse dependency
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/sandboxfs/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "bazelbuild";
+    repo = pname;
+    rev = "sandboxfs-${version}";
+    sha256 = "Ia6rq6FN4abnvLXjlQh4Q+8ra5JThKnC86UXC7s9//U=";
+  };
+
+  cargoSha256 = "sha256-k303TjWG+n+/ZMmko84KJtYb7swuQ1ZJOc4Vq6aOhX0=";
```
Comment (danieldk) [inline pkgs/tools/filesystems/sandboxfs/default.nix:20]: I always forget if we already permit SRI hashes, but from a quick grep it seems that we have a lot of them already.

File: pkgs/tools/filesystems/sandboxfs/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  cargoPatches = [ ./Cargo.lock.patch ];
```
Comment (danieldk) [inline pkgs/tools/filesystems/sandboxfs/default.nix:22]: Could you open an issue upstream asking them to add `Cargo.lock` and add a comment here with the URL of the issue?

File: pkgs/tools/filesystems/sandboxfs/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = [ fuse ];
+  nativeBuildInputs = [ pkg-config installShellFiles ];
```
Comment (danieldk) [inline pkgs/tools/filesystems/sandboxfs/default.nix:25]: Nitpick, but this is the canonical order ;).

```suggestion
  nativeBuildInputs = [ pkg-config installShellFiles ];

  buildInputs = [ fuse ];
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks, looks good! Two small requested changes.

File: pkgs/tools/filesystems/sandboxfs/default.nix (hunk 3 above)
Comment (jeremyschlatter) [reply pkgs/tools/filesystems/sandboxfs/default.nix]: Done.

File: pkgs/tools/filesystems/sandboxfs/default.nix (hunk 2 above)
Comment (jeremyschlatter) [reply pkgs/tools/filesystems/sandboxfs/default.nix:23]: Done.

File: pkgs/tools/filesystems/sandboxfs/default.nix (hunk 1 above)
Comment (jeremyschlatter) [reply pkgs/tools/filesystems/sandboxfs/default.nix:20]: I computed these hashes by setting these fields to the empty string and then copy-pasting the hash in the error message from nix-build. This is the format that nix-build used to print them. I'd be happy to switch to a different format if that's preferable.

**Comment by jeremyschlatter** [comment]
Thanks for the review! I made those two small changes.

**Review by danieldk — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 99353` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sandboxfs</li>
  </ul>
</details>

**Pull request merged**